### ISSUE TITLE
Winding order

### ISSUE THREAD

**Issue by jeannekamikaze** [HBehrens/obj2opengl#4]
Unless I am mistaken, it seems that the second triangle generated from a quad should be ACD, not ADC. If a quad's vertices are given as A B C D, the two triangles generated would be ABC and ACD, otherwise half of the triangles end up clockwise and the other half counter-clockwise and then face culling fails to work properly.

``` perl
if($5 != "")
        {
            @d = split('/', $5);
            $va_idx[$numFaces] = $a[0]-1;
            $ta_idx[$numFaces] = $a[1]-1;
            $na_idx[$numFaces] = $a[2]-1;

            $vb_idx[$numFaces] = $c[0]-1;
            $tb_idx[$numFaces] = $c[1]-1;
            $nb_idx[$numFaces] = $c[2]-1;

            $vc_idx[$numFaces] = $d[0]-1;
            $tc_idx[$numFaces] = $d[1]-1;
            $nc_idx[$numFaces] = $d[2]-1;

            $face_line[$numFaces] = $line;

            $numFaces++;
        }
```

**Comment by patr1ck** [comment]
Hey there, inexperienced OpenGL newbie here. In my project, the objects i'm using (converted from blender .obj files to .h files using this script) are covered in a triangle checker pattern, where I half the checkered parts are completely transparent.

I can fix the problem by removing `glEnable(GL_CULL_FACE);`, but that seems incorrect.

Sounds like I'm hitting the above issue?

**Comment by patr1ck** [comment]
Yep, I tried the above fix and now things work fine even with GL_CULL_FACE. Thanks @jeannekamikaze!
